modules/tinymce/src/themes/silver/main/ts/ui/core/ChoiceControls.ts

```diff
@@ -37,14 +37,23 @@ const registerController = <T>(editor: Editor, spec: ControlSpec<T>) => {
     const options = spec.getOptions(editor);
     const initial = spec.getCurrent(editor).map(spec.hash);
 
+    const current = Singleton.destroyable();
+
     return Arr.map(
       options,
-      (value, i) => ({
+      (value) => ({
         type: 'togglemenuitem',
         text: spec.display(value),
         onSetup: (api) => {
-          api.setActive(Optionals.is(initial, spec.hash(value)));
-          return spec.watcher(editor, value, api.setActive);
+          const setActive = (active: boolean) => {
+            api.setActive(active);
+            if (active) {
+              current.set({ destroy: () => api.setActive(false) });
```

Comment:
After a _significant_ amount of time... I now understand what this is doing. Singletons have always been a bit odd for me, but do you think we could do it a slightly more verbose but easier to understand way using `Singleton.value`?

```
current.on((oldApi) => oldApi.setActive(false));
current.set(api);
```